linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -45,6 +45,7 @@ public class MarkStep : IStep {
 		protected List<MethodDefinition> _virtual_methods;
 		protected Dictionary<TypeDefinition, CustomAttribute> _assemblyDebuggerDisplayAttributes;
 		protected Dictionary<TypeDefinition, CustomAttribute> _assemblyDebuggerTypeProxyAttributes;
+		protected Queue<CustomAttribute> _lazyAttributes;
```

Comment:
Do you not see this scheme being useful to attributes on other elements?  I kept the naming generic because I thought we may want to process other attributes after all types have been processed.